### DISCUSSION TITLE
is Open-wrt 19.07.07 needed? Or could a newer version suffice?

### DISCUSSION THREAD

**Discussion by N01Important** [xchwarze/wifi-pineapple-cloner#6] [Q&A] [answered]
I'm running into issues getting 19.07.7 installed via any method short of jtag. If the requirement is any version of open-wrt it would be easier.

Device: AR-750S (gl inet slate)

**Comment by xchwarze** [comment]
hi pal!
I understand that in all glinet routers OpenWrt is installed in the same way, always via recovery.
The original project repo was deleted by h5... so the detailed explanations were lost.

**Reply by N01Important** [reply]
Thanks for reaching back out.

I'm partially successful! As in I have openwrt running but its 23.05 vs 19.07. Hopefully that won't be a huge issue but we'll see. 

in case anyone else stumbles across this for the ar-750s-ext:

* uboot and install the nor version of the firmware - openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-**nor**-squashfs-sysupgrade.bin

* that will reboot and the power button will show steady green

* luci web gui firmware update to the nor-nand firmware -openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-**nor-nand**-squashfs-sysupgrade.bin


That's where I am now, I'll try later to get a ssh then firmware update with the cloner firmware and see where we end up.

**Reply by xchwarze** [reply]
As glinet routers are already made with OpenWrt the only thing you have to do is to enter recovery and upload the project firmware.

these should be the steps https://docs.gl-inet.com/router/en/3/tutorials/debrick/

**Reply by N01Important** [reply]
The issue is the uboot on my AR750S-Ext seems to be the newer version and it won't take the pineapple firmware.

**Reply by N01Important** [reply]
you address all of this in your Patreon posts don't you? :)

**Reply by xchwarze** [reply]
Strange because since this router is discontinued it should be the same version that we all bought and use correctly.

Anyway if you have already installed openwrt 22/23 follow the installation steps in the manual and you should be able to install correctly.

**Comment by N01Important** [comment, marked as answer]
Hey thank you for the assistance. I got it working.

Step 1: U-boot and upload **NOR ONLY** openwrt image, I did with  openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-**nor**-squashfs-sysupgrade.bin.

Step 2: upload the pineapple cloner firmware using the install instructions

Step 3: Setup your new pineapple-cloner flashed device

**Reply by tylergates-dev** [reply]
Which version of the pineapple cloner did you use? 19.07.07? Or did you have to custom build one using the build from scratch instructions?

**Reply by xchwarze** [reply]
AR750s is the router I recommend to buy and therefore the one I have to test. The only thing you have to do is enter recovery mode (the page accessed from uboot) and upload the firmware.
On the OpenWRT page there are several notes on how to install it on this router https://openwrt.org/toh/gl.inet/gl-ar750s

**Reply by tylergates-dev** [reply]
> AR750s is the router I recommend to buy and therefore the one I have to test. The only thing you have to do is enter recovery mode (the page accessed from uboot) and upload the firmware. On the OpenWRT page there are several notes on how to install it on this router https://openwrt.org/toh/gl.inet/gl-ar750s

How do you enter UBoot?  I have followed the steps on every guide and can't get it into whatever UBoot mode is. I'm on win11. I have bricked it so many times (and unbricked it) that now I get stuck in boot loops where I get to Luci but nothing else. How do you get into boot? I've changed to the static ip and everything else (following instructions) and I can't get any other screen other than Luci.

**Reply by xchwarze** [reply]
https://docs.gl-inet.com/router/en/3/tutorials/debrick/
```
For GL-AR150, GL-AR300M, GL-USB150(Microuter), GL-AR750(Creta), GL-AR750S-EXT(Slate), GL-X750(Spitz), GL-MT300N-V2(Mango), microuter-N300, the LED falshes 5 times.
```

**Reply by tylergates-dev** [reply]
> https://docs.gl-inet.com/router/en/3/tutorials/debrick/
> 
> ```
> For GL-AR150, GL-AR300M, GL-USB150(Microuter), GL-AR750(Creta), GL-AR750S-EXT(Slate), GL-X750(Spitz), GL-MT300N-V2(Mango), microuter-N300, the LED falshes 5 times.
> ```

This is the guide I have followed. It's now in a boot loop. That's what I'm trying to explain lol

**Reply by xchwarze** [reply]
Then I followed the steps of N01Important that from what he says it worked for him. I didn't need any more steps than that.

**Reply by tylergates-dev** [reply]
> Then I followed the steps of N01Important that from what he says it worked for him. I didn't need any more steps than that.

Interesting. so here is my issue. I am currently on Firmware 23.05.00 NOR/NAND and im fully unable to flash 19.07.07 (As required). Any time i do i run into the same problem everyone else is having where it just reboots and nothing changes. Issue 2: I am fully unable to get to the "Uboot" screen shown in all of these recovery tutorials. I am in the router at this point on that 23.x.x firmware but thats as far as i can get at this point. 

Sorry for being dumb with this one but man am i struggling lol. 

Also, Im unable to install anything older than 23.05.00 because of the NOR/NAND firmware being the only flavor of firmware that can be flashed. Im unable to flash straight NOR or Straight NAND firmware, only the combo. I feel like im missing something

[backup-OpenWrt-2024-09-23.tar.gz](https://github.com/user-attachments/files/18025067/backup-OpenWrt-2024-09-23.tar.gz)
![Screenshot 2024-12-05 100117](https://github.com/user-attachments/assets/630c28be-0a44-4330-904e-22089cfb1d90)
![Screenshot 2024-12-05 100324](https://github.com/user-attachments/assets/ed977946-5e00-4073-9710-954e0f716f25)
![Screenshot 2024-12-05 101500](https://github.com/user-attachments/assets/e32c1ee9-e15b-4cfb-935e-2b114f59e5a9)



Hopefully im just being stupid and am missing something but i have followed every single link and tutorial to a T and still am struggling here

**Reply by tylergates-dev** [reply]
> Then I followed the steps of N01Important that from what he says it worked for him. I didn't need any more steps than that.

The steps that 

> Hey thank you for the assistance. I got it working.
> 
> Step 1: U-boot and upload **NOR ONLY** openwrt image, I did with openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-**nor**-squashfs-sysupgrade.bin.
> 
> Step 2: upload the pineapple cloner firmware using the install instructions
> 
> Step 3: Setup your new pineapple-cloner flashed device

Were listed here. When you follow install instructions are you talking about the custom build? Or are you uploading and Installing the 19.07.07 prebuild? Also, as previously stated I'm using Win11. I also have a machine running fadora, would that be better to use?

**Reply by xchwarze** [reply]
In OpenWRT 19 this router has ONLY NOR support. You don't need to custom build anything. 
At most for what is happening you might have to install openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-nor-squashfs-sysupgrade.bin and go from there.

I don't know what else to tell you because it really should have worked for you with uboot. Maybe your router is defective.

**Reply by tylergates-dev** [reply]
> In OpenWRT 19 this router has ONLY NOR support. You don't need to custom build anything. At most for what is happening you might have to install openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-nor-squashfs-sysupgrade.bin and go from there.
> 
> I don't know what else to tell you because it really should have worked for you with uboot. Maybe your router is defective.

Okay. makes sense. might pick up another one. Now I see two different ways of installing, First is using CMD and running the SCP Commands. The other is going through the user interface / 192.168.1.1 (im talking for the Pineapple firmware. Which do you recommend here?

**Reply by tylergates-dev** [reply]
> In OpenWRT 19 this router has ONLY NOR support. You don't need to custom build anything. At most for what is happening you might have to install openwrt-23.05.0-ath79-nand-glinet_gl-ar750s-nor-squashfs-sysupgrade.bin and go from there.
> 
> I don't know what else to tell you because it really should have worked for you with uboot. Maybe your router is defective.

You guys have been a big help! I finally got it to work!! My MAIN Tips I learned for anyone struggling is to make sure you clear browser cookies, be patient, and follow the above directions. Also, once you flash the pineapple firmware, to set it up connect to the pineapple wifi and not through ethernet. It might not show up via ethernet but should be showing it's wifi
